### PR TITLE
Add support for CH for instance status reports

### PR DESCRIPTION
**Pull request by fuziontech**
## Changes

This adds some signal for how many events clickhouse on prem instances have

Example payload:
```json
instance status report {"posthog_version": "1.28.0", "deployment": "unknown", "realm": "hosted-clickhouse", "period": {"start_inclusive": "2021-09-06T00:00:00+00:00", "end_inclusive": "2021-09-12T23:59:59.999999+00:00"}, "site_url": "unknown", "license_keys": [], "helm": {}, "users_who_logged_in": [{"id": 1, "distinct_id": "DJWPQYiUJydofPLrAKndnOiSWTwbVaM4XT6ldLxIAbj", "first_name": "test", "email": "test@test.com"}], "teams": {"1": {"events_count_total": 12, "events_count_new_in_period": 0, "events_count_by_lib": {}, "events_count_by_name": {}, "persons_count_total": 1, "persons_count_new_in_period": 0, "persons_count_active_in_period": 0, "dashboards_count": 1, "dashboards_template_count": 0, "dashboards_shared_count": 0, "dashboards_tagged_count": 0, "ff_count": 0, "ff_active_count": 0}, "2": {"events_count_total": 1876, "events_count_new_in_period": 694, "events_count_by_lib": {"": 588, "web": 106}, "events_count_by_name": {"purchase": 7, "$pageview": 347, "watched_movie": 97, "installed_app": 109, "rated_app": 81, "$autocapture": 53}, "persons_count_total": 321, "persons_count_new_in_period": 0, "persons_count_active_in_period": 0, "dashboards_count": 3, "dashboards_template_count": 0, "dashboards_shared_count": 0, "dashboards_tagged_count": 0, "ff_count": 0, "ff_active_count": 0}}, "table_sizes": {"posthog_event": 57344, "posthog_sessionrecordingevent": 49152}, "plugins_installed": {"GeoIP": 2}, "plugins_enabled": {"GeoIP": 2}, "instance_usage_summary": {"events_count_new_in_period": 694, "persons_count_new_in_period": 0, "persons_count_total": 322, "events_count_total": 1888, "dashboards_count": 4, "ff_count": 0}}
{'posthog_version': '1.28.0', 'deployment': 'unknown', 'realm': 'hosted-clickhouse', 'period': {'start_inclusive': '2021-09-06T00:00:00+00:00', 'end_inclusive': '2021-09-12T23:59:59.999999+00:00'}, 'site_url': 'unknown', 'license_keys': [], 'helm': {}, 'users_who_logged_in': [{'id': 1, 'distinct_id': 'DJWPQYiUJydofPLrAKndnOiSWTwbVaM4XT6ldLxIAbj', 'first_name': 'test', 'email': 'test@test.com'}], 'teams': {1: {'events_count_total': 12, 'events_count_new_in_period': 0, 'events_count_by_lib': {}, 'events_count_by_name': {}, 'persons_count_total': 1, 'persons_count_new_in_period': 0, 'persons_count_active_in_period': 0, 'dashboards_count': 1, 'dashboards_template_count': 0, 'dashboards_shared_count': 0, 'dashboards_tagged_count': 0, 'ff_count': 0, 'ff_active_count': 0}, 2: {'events_count_total': 1876, 'events_count_new_in_period': 694, 'events_count_by_lib': {'': 588, 'web': 106}, 'events_count_by_name': {'purchase': 7, '$pageview': 347, 'watched_movie': 97, 'installed_app': 109, 'rated_app': 81, '$autocapture': 53}, 'persons_count_total': 321, 'persons_count_new_in_period': 0, 'persons_count_active_in_period': 0, 'dashboards_count': 3, 'dashboards_template_count': 0, 'dashboards_shared_count': 0, 'dashboards_tagged_count': 0, 'ff_count': 0, 'ff_active_count': 0}}, 'table_sizes': {'posthog_event': 57344, 'posthog_sessionrecordingevent': 49152}, 'plugins_installed': Counter({'GeoIP': 2}), 'plugins_enabled': Counter({'GeoIP': 2}), 'instance_usage_summary': {'events_count_new_in_period': 694, 'persons_count_new_in_period': 0, 'persons_count_total': 322, 'events_count_total': 1888, 'dashboards_count': 4, 'ff_count': 0}}
```
## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: ee/clickhouse/models/person.py [hunk 1]
```diff
@@ -18,12 +18,11 @@
 )
 from ee.kafka_client.client import ClickhouseProducer
 from ee.kafka_client.topics import KAFKA_PERSON, KAFKA_PERSON_UNIQUE_ID
-from posthog import settings
 from posthog.models.person import Person, PersonDistinctId
 from posthog.models.utils import UUIDT
 from posthog.utils import is_clickhouse_enabled
 
-if settings.EE_AVAILABLE and is_clickhouse_enabled():
+if is_clickhouse_enabled():
```
Comment (fuziontech) [inline ee/clickhouse/models/person.py:25]: This is a cleanup as `is_clickhouse_enabled` already checks if `settings.EE_AVAILABLE` is `True`

File: ee/clickhouse/models/event.py [hunk 2]
```diff
@@ -202,3 +201,69 @@ def determine_event_conditions(
...
+
+def get_event_count_for_team_and_period(
+    team_id: Union[str, int], begin: timezone.datetime, end: timezone.datetime
+) -> int:
+    result = sync_execute(
+        """
+        SELECT count(1) as count
+        FROM events
+        WHERE team_id = %(team_id)s
+        AND timestamp > %(begin)s
```
Comment (kpthatsme) [inline ee/clickhouse/models/event.py:214]: I think we should use >= / <= here so it's an inclusive period?

Comment (kpthatsme) [reply ee/clickhouse/models/event.py:214]: or maybe even better, could we use `between`?

**Review by kpthatsme — COMMENTED** [review]
awesome @fuziontech just one comment

File: posthog/tasks/status_report.py [hunk 3]
```diff
@@ -63,31 +69,57 @@ def status_report(*, dry_run: bool = False) -> Dict[str, Any]:
...
+                    get_events_count_for_team_by_client_lib,
+                    get_events_count_for_team_by_event_type,
+                )
+
+                team_event_count = get_event_count_for_team(team.id)
+                instance_usage_summary["events_count_total"] += team_event_count
+                team_report["events_count_total"] = team_event_count
+                team_events_in_period_count = get_event_count_for_team_and_period(team.id, period_start, period_end)
+                team_report["events_count_new_in_period"] = team_events_in_period_count
+                instance_usage_summary["events_count_new_in_period"] += team_report["events_count_new_in_period"]
```
Comment (tiina303) [inline posthog/tasks/status_report.py:88]:
```suggestion
                instance_usage_summary["events_count_new_in_period"] += team_events_in_period_count
```
nit for readability/consistency

File: posthog/tasks/status_report.py [hunk 4]
```diff
@@ -63,31 +69,57 @@ def status_report(*, dry_run: bool = False) -> Dict[str, Any]:
...
+                )
+            else:
+                # pull events stats from postgres
+                events_considered_total = Event.objects.filter(team_id=team.id)
+                instance_usage_summary["events_count_total"] += events_considered_total.count()
+                events_considered_new_in_period = events_considered_total.filter(
+                    timestamp__gte=period_start, timestamp__lte=period_end,
+                )
+
+                team_report["events_count_total"] = events_considered_total.count()
```
Comment (tiina303) [inline posthog/tasks/status_report.py:104]: n00b Q: do we run the query when we call `count()` if so we are doing `events_considered_total.count()` twice unnecessarily here, we could use a variable like you did for clickhouse part or just take it from the `team_report` for `instance_usage_summary` like we do below. Also a consistency nit.

File: ee/clickhouse/models/event.py [hunk 5]
```diff
@@ -202,3 +201,69 @@ def determine_event_conditions(
...
+        SELECT count(1) as count
+        FROM events
+        WHERE team_id = %(team_id)s
+    """,
+        {"team_id": str(team_id)},
+    )[0][0]
+    return result
+
+
+def get_event_count() -> int:
```
Comment (tiina303) [inline ee/clickhouse/models/event.py:234]: nit: not used anywhere, but we can keep it

File: ee/clickhouse/models/event.py [hunk 6]
```diff
@@ -202,3 +201,69 @@ def determine_event_conditions(
...
+def get_event_count_for_team_and_period(
+    team_id: Union[str, int], begin: timezone.datetime, end: timezone.datetime
+) -> int:
+    result = sync_execute(
+        """
+        SELECT count(1) as count
+        FROM events
+        WHERE team_id = %(team_id)s
+        AND timestamp > %(begin)s
+        AND timestamp < %(end)s
```
Comment (tiina303) [inline ee/clickhouse/models/event.py:215]: this doesn't match what we do for postgres, which is <= and >= based on what I assume `timestamp__gte` means (here it's < & >). It also doesn't match the queries below.

File: ee/clickhouse/models/event.py (hunk 5 above)
Comment (fuziontech) [reply ee/clickhouse/models/event.py:234]: Yeah it's something we can use elsewhere

File: posthog/tasks/status_report.py (hunk 4 above)
Comment (fuziontech) [reply posthog/tasks/status_report.py:104]: love it but not going to spend any time on Postgres code right now. Can't wait to 🔥 it all 😈

**Pull request merged**